src/auth/auth.service.ts

```diff
@@ -1,14 +1,17 @@
-import { ConflictException, Injectable, UnauthorizedException } from '@nestjs/common';
-import { JwtService } from '@nestjs/jwt';
+import { ConflictException, forwardRef, Inject, Injectable, UnauthorizedException } from '@nestjs/common';
 import { UserCredentialsDto, UserDto, UserInfoDto } from '../user/dto/user.dto';
+import { JwtService } from '@nestjs/jwt';
 import { RefreshTokenDto, TokensDto } from './dto/tokens.dto';
 import { UserService } from '../user/user.service';
 
 import bcrypt from 'bcrypt';
 
 @Injectable()
 export class AuthService {
-  constructor(private userService: UserService, private jwtService: JwtService) {}
+  constructor(
+    @Inject(forwardRef(() => UserService)) private userService: UserService,
```

Comment:
-> you don't need here 
https://docs.nestjs.com/fundamentals/circular-dependency